zjit/src/hir.rs

```diff
@@ -2443,6 +2446,48 @@ impl Function {
                                     self.push_insn(block, Insn::LoadField { recv: as_heap, id: mid, offset, return_type: types::BasicObject })
                                 };
                                 self.make_equal_to(insn_id, replacement);
+                            } else if let (OPTIMIZED_METHOD_TYPE_STRUCT_ASET, &[val]) = (opt_type, args.as_slice()) {
+                                if unsafe { vm_ci_argc(ci) } != 1 {
```

Comment:
Is this redundant with the `&[val]` pattern matching above and can it be removed?